openshift-hack/rebase.sh

```diff
@@ -0,0 +1,171 @@
+  echo "  --bugzilla-id        (optional) creates new PR against openshift/kubernetes:${openshift-release}: Example: --bugzilla-id=2003027"
+}
+
+for i in "$@"; do
+  case $i in
+  --k8s-tag=*)
+    k8s_tag="${i#*=}"
+    shift
+    ;;
+  --openshift-release=*)
+    openshift_release="${i#*=}"
+    shift
+    ;;
+  --bugzilla-id=*)
+    bugzilla_id="${i#*=}"
+    shift
+    ;;
+  *)
+    usage
+    exit 1
+    ;;
+  esac
+done
+
+if [ -z "${k8s_tag}" ]; then
+  echo "Required argument missing: --k8s-tag"
+  echo ""
+  usage
+  exit 1
+fi
+
+if [ -z "${openshift_release}" ]; then
+  echo "Required argument missing: --openshift-release"
+  echo ""
+  usage
+  exit 1
+fi
+
+echo "Processed arguments are:"
+echo "--k8s_tag=${k8s_tag}"
+echo "--openshift_release=${openshift_release}"
+echo "--bugzilla_id=${bugzilla_id}"
+
+# prerequisites (check git, podman, ... is present)
+if ! command -v git &>/dev/null; then
+  echo "git not installed, exiting"
+  exit 1
+fi
+
+if ! command -v jq &>/dev/null; then
+  echo "jq not installed, exiting"
+  exit 1
+fi
+
+if ! command -v podman &>/dev/null; then
+  echo "podman not installed, exiting"
+  exit 1
+fi
+
+# make sure we're in "kubernetes" dir
+if [[ $(basename "$PWD") != "kubernetes" ]]; then
+  echo "Not in kubernetes dir, exiting"
+  exit 1
+fi
+
+origin=$(git remote get-url origin)
+if [[ "$origin" =~ .*kubernetes/kubernetes.* || "$origin" =~ .*openshift/kubernetes.* ]]; then
+  echo "cannot rebase against k/k or o/k! found: ${origin}, exiting"
+  exit 1
+fi
+
+# fetch remote https://github.com/kubernetes/kubernetes
+git remote add upstream git@github.com:kubernetes/kubernetes.git
+git fetch upstream --tags -f
+# fetch remote https://github.com/openshift/kubernetes
+git remote add openshift git@github.com:openshift/kubernetes.git
+git fetch openshift
+
+git checkout --track "openshift/$openshift_release"
+git pull openshift "$openshift_release"
+
+git merge "$k8s_tag"
+# shellcheck disable=SC2181
+if [ $? -eq 0 ]; then
+  echo "No conflicts detected. Automatic merge looks to have succeeded"
+else
+  # commit conflicts
+  git commit -a
+  # resolve conflicts
+  git status
+  # TODO(tjungblu): we follow-up with a more automated approach:
+  # - 2/3s of conflicts stem from go.mod/sum, which can be resolved deterministically
+  # - the large majority of the remainder are vendor/generation conflicts
+  # - only very few cases require manual intervention due to conflicting business logic
+  echo "Resolve conflicts manually in another terminal, only then continue"
+
+  # wait for user interaction
+  read -n 1 -s -r -p "PRESS ANY KEY TO CONTINUE"
+
+  # TODO(tjungblu): verify that the conflicts have been resolved
+  git commit -am "UPSTREAM: <drop>: manually resolve conflicts"
+fi
+
+# update dependencies
+sed -i -E "/=>/! s/(\tgithub.com\/openshift\/[a-z|-]+) (.*)$/\1 $openshift_release/" go.mod
+go mod tidy
+
+hack/update-vendor.sh
+# shellcheck disable=SC2181
```

Comment:
go mod and the vendor update can potentially take a higher go version locally, we need to use podman for the invocation too.

Reply:
Yeah, `podman run -it --rm -v $( pwd ):/go/k8s.io/kubernetes:Z --workdir=/go/k8s.io/kubernetes registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.16-openshift-4.9 make update OS_RUN_WITHOUT_DOCKER=yes` with appropriate golang and openshift release is the best option you have as mentioned elsewhere in this doc.

Reply:
Oh, I misread, yeah, for vendor update maybe it's also worth using the same podman invocation as for `make update`.

Reply:
yep, exactly. This is being done below a couple of lines, but that's how the build tags remained ultimately.